src/s390_ecc.c

```diff
@@ -1743,16 +1744,26 @@ struct {				\
 #ifdef ICA_FIPS
 			if (fips & ICA_FIPS_MODE) {
 				fc |= 0x80; /* deterministic signature */
-				RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+				do {
+					/* A cc=2 occurs if the random number is not invertible.
+					 * Loop in this case. Same for p384 and p521 below. */
+					counter++;
+					if (RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off) != 1) {
+						rc = EIO;
+						break;
+					}
+					rc = s390_kdsa(fc, param.buff, NULL, 0);
+				} while (rc != 0 && counter < MAX_KDSA_RETRIES);
+			} else {
+				rc = s390_kdsa(fc, param.buff, NULL, 0);
```

Comment:
s390_kdsa returns 0 on success and 1 on error, and we return it as is from ecdsa_sign_cpacf. However, isn't ecdsa_sign_cpacf supposed to return an errno value in case of an error. 1 is EPERM (Operation not permitted).... is this what we want to return? Maybe we should change s390_kdsa to initialize rc to EIO instead? EINVAL could fit better (i.e. invalid deterministic nonce), but EINVAL means curve not supported as per caller  ecdsa_sign_hw....

Reply:
ok, done.